src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -38,6 +38,37 @@ internal abstract partial class AbstractReferenceFinder : IReferenceFinder
         public abstract ValueTask<ImmutableArray<FinderLocation>> FindReferencesInDocumentAsync(
             ISymbol symbol, FindReferencesDocumentState state, FindReferencesSearchOptions options, CancellationToken cancellationToken);
 
+        protected virtual ValueTask<(bool matched, CandidateReason reason)> SymbolsMatchAsync(
+            ISymbol symbol, FindReferencesDocumentState state, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            // delegates don't have exposed symbols for their constructors.  so when you do `new MyDel()`, that's only a
+            // reference to a type (as we don't have any real constructor symbols that can actually cascade to).  So
+            // don't do any special finding in that case.
+            var parent = symbol.IsDelegateType()
+                ? token.Parent
+                : state.SyntaxFacts.TryGetBindableParent(token);
+            parent ??= token.Parent!;
+
+            return SymbolsMatchAsync(symbol, state, parent, cancellationToken);
+        }
+
+        protected static async ValueTask<(bool matched, CandidateReason reason)> SymbolsMatchAsync(
```

Comment:
renamed as this is the same as the above method, just taking a node instead of a token.